search/src/java/cz/incad/Kramerius/oai/OaiServlet.java

```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         try {
             sendResponse(oaiWriter, resp);
         } catch (IOException ex) {
-            LOG.log(Level.SEVERE, pid, ex);
+            LOG.log(Level.WARNING, pid + " doesn't exist in Fedora.");
             if (!resp.isCommitted()) {
                 resp.reset();
-                resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                sendFakeContent(resp);
             }
         } finally {
             oaiWriter.close();
         }
 
     }
+    
+    private void sendFakeContent(HttpServletResponse resp) {
```

Comment:
přejmenoval bych na sendErrorMessage

Reply:
Nedával bych tam code="404", protože ty chyby se můžou lišit....lepší bude mít atribut message a před tam message z IOException. Pokud IOEXception nenese rozumnou hlášku bude pravděpodobně nutné upravit DrKrameriusV4Writer.